config/url.go

```diff
@@ -17,7 +17,7 @@ type URL struct {
 	Name string
 	// Any URL that starts with this value will be rewritten to start, instead, with <base>.
 	// When more than one insteadOf strings match a given URL, the longest match is used.
-	InsteadOf string
+	InsteadOf []string
```

Comment:
Please add to the commit body (and the PR description) that this introduces a breaking change to `config.URL`.

Reply:
Should i make this an `interface` (with corresponding type casting) to try to preserve compatibility here?

I know version #s are free, but it seems a little silly to bump the major for only this...

Reply:
@stensonb I agree that the bump may not be worth the change.

@BryanStenson-okta I would be onboard with adding an additional `InsteadOfSlices []string` field. We would have to support both until `v6`, when we would drop the new field and change the type of the existing. What do you think?

The new field would have a deprecation note as we have in [other places around the code](https://github.com/go-git/go-git/blob/6252084d6fd3173aceeecd6765722ae844ddb266/worktree_status.go#L273).